tests/test-mongoc-cursor.c

```diff
@@ -733,6 +733,80 @@ test_cursor_new_invalid (void)
    mongoc_client_destroy (client);
 }
 
+
+static void
+test_cursor_new_invalid_filter (void)
+{
+   mock_server_t *server;
+   mongoc_client_t *client;
+   mongoc_collection_t *collection;
+   mongoc_cursor_t *cursor;
+   bson_error_t error;
+
+   server = mock_server_new ();
+   mock_server_run (server);
+   client = mongoc_client_new_from_uri (mock_server_get_uri (server));
```

Comment:
Don't need a mock server and all that ceremony - you can just do client_new and then find_with_opts.

(For future reference: mock server is useful to simulating server errors that are hard to trigger with a real server, or for reading messages sent *from* the driver and validating their contents. If you *do* need a mock server for such a scenario later on, don't forget to mock_server_destroy!)